hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/pipeline/Pipeline.java

```diff
@@ -50,6 +53,8 @@
   private Map<DatanodeDetails, Long> nodeStatus;
   // nodes with ordered distance to client
   private ThreadLocal<List<DatanodeDetails>> nodesInOrder = new ThreadLocal<>();
+  // Current reported Leader for the pipeline
+  private UUID leaderId;
```

Comment:
nitpick: We can make it strongly typed by making the type as `DatanodeDetails`

Reply:
But if I do that, in order to construct Pipeline from proto, I have to lookup DatanodeDetails based on UUID, which is a map in NodeManager. Not sure why making this into DatanodeDetails is better?

Reply:
It is just that it will be explicit what this UUID represent.